cogs/general/general.py

```diff
@@ -0,0 +1,433 @@
+            await ctx.send("I need the `Embed links` permission "
+                            "to send this")
+
+    @commands.command()
+    @commands.guild_only()
+    async def guildinfo(self, ctx):
+        """Shows guild's informations"""
+        guild = ctx.message.guild
+        online = len([m.status for m in guild.members
+                      if m.status == discord.Status.online or
+                      m.status == discord.Status.idle])
+        total_users = len(guild.members)
+        text_channels = len([x for x in guild.channels
+                             if x.type == discord.ChannelType.text])
+        voice_channels = len(guild.channels) - text_channels
+        passed = (ctx.message.timestamp - guild.created_at).days
+        created_at = ("Since {}. That's over {} days ago!"
+                      "".format(guild.created_at.strftime("%d %b %Y %H:%M"),
+                                passed))
+
+        colour = ''.join([choice('0123456789ABCDEF') for x in range(6)])
+        colour = int(colour, 16)
+
+        data = discord.Embed(
+            description=created_at,
+            colour=discord.Colour(value=colour))
+        data.add_field(name="Region", value=str(guild.region))
+        data.add_field(name="Users", value="{}/{}".format(online, total_users))
+        data.add_field(name="Text Channels", value=text_channels)
+        data.add_field(name="Voice Channels", value=voice_channels)
+        data.add_field(name="Roles", value=len(guild.roles))
+        data.add_field(name="Owner", value=str(guild.owner))
+        data.set_footer(text="Guild ID: " + guild.id)
+
+        if guild.icon_url:
+            data.set_author(name=guild.name, url=guild.icon_url)
+            data.set_thumbnail(url=guild.icon_url)
+        else:
+            data.set_author(name=guild.name)
+
+        try:
+            await ctx.send(embed=data)
+        except discord.HTTPException:
+            await ctx.send("I need the `Embed links` permission "
+                            "to send this")
+
+    @commands.command()
+    async def urban(self, ctx, *, search_terms: str, definition_number: int=1):
+        """Urban Dictionary search
+
+        Definition number must be between 1 and 10"""
+        def encode(s):
+            return quote_plus(s, encoding='utf-8', errors='replace')
+
+        # definition_number is just there to show up in the help
+        # all this mess is to avoid forcing double quotes on the user
+
+        search_terms = search_terms.split(" ")
+        try:
+            if len(search_terms) > 1:
+                pos = int(search_terms[-1]) - 1
+                search_terms = search_terms[:-1]
+            else:
+                pos = 0
+            if pos not in range(0, 11): # API only provides the
+                pos = 0                 # top 10 definitions
+        except ValueError:
+            pos = 0
+
+        search_terms = "+".join([encode(s) for s in search_terms])
```

Comment:
Honestly this entire function is fucked.

`aiohttp` supports `params` as a keyword argument and will automatically do this for you. I don't even know what's going on here.